### DISCUSSION TITLE
Track recording dialog

### DISCUSSION THREAD

**Discussion by kreuzschnabel** [osmandapp/OsmAnd#11958] [General]
Look here, OsmAnd, when I hit the "REC" icon in the panel (maybe when I’m driving), I just want you to start recording a track silently in the background while still showing me the map. I do NOT want to firstly answer umpteen popping-up dialogs about which colour to display it or maybe select a different tracking interval than last time or if i want some extra cheese on it. This is not just annoying, this is even dangerous when driving. Just start tracking when I tap the icon as you did until yesterday, and just do it all the same way you did it last time. I’ll tell you in time when I want something about it done differently. There’s no need to ask all that EVERY F***ING TIME ANEW when I just want you to start recording while I’m paying attention to something else. Thank you.

**Comment by pebogufi** [comment]
FULLY agree !

**Comment by pebogufi** [comment]
See also #10924

**Comment by tommycrock** [comment]
Well, I don't agree with kreuzschnabel's tone but, I do agree that, at least the second dialogue is wholly unnecessary to pop up when starting recording. The first dialogue should ideally be optional from the widget, maybe shown at first ever recording and then when recording via menu and its future showing being an offer in the current recording track dialogue. By the way, it's great that the current recording track dialogue has been improved to allow switching the visibility.

**Comment by Ungoose** [comment]
If I remember correctly, switching the visibility was present before the dialog was redesigned - it just had a "Do not ask again" checkbox, so the user could configure it on the first launch and never see it again. There is an "always ask" toggle in the new menu but it doesn't seem to work - the menu pops up regardless of the setting, as mentioned in #10924

**Comment by tommycrock** [comment]
I don't remember it being an option while recording, but could be wrong, and found it switched of its own accord based on on whether I'd wanted to view the finished track from the previous recording. Then I had to find a way to turn it back on while recording: My places, tracks, currently recording track was the only place I could find it, which didn't seem obvious to me.

**Comment by Ungoose** [comment]
Ah, yes, you are right. Now the option is available both in "Start recording" dialog (where you choose the logging interval) AND in the new recording dialog/dashboard (where you can pause/stop/clear the track). That is an improvement.

**Comment by brapamaldi** [comment]
> this is even dangerous when driving

in my country touching a phone screen while driving is illegal, regardless if its annoying extra touches or not. maybe pull over and press record then drive and not be a hazard on the road. nothing annoys me more than almost being hit by fools using their phones while driving

**Reply by Brucetld** [reply]
My answer to all automotive touch screens is a is a track ball or mouse (on your knee).  Stops the distracted driving thing.  Unfortunately,  car mfrs don't understand or offer this.  OSMand does,  buy a Bluetooth model, used one for years.  Like old car radios, you can use it by feeling, not looking!  BR

**Comment by frasty** [comment]
I agree with @kreuzschnabel . Track recording dialogs are irritating either when starting or *stopping* a track. A major use of Osmand is outdoor orienteering and all these confirmation dialogs are nonsense especially when you use the phone in outdoor situations with poor contrast due to direct sunlight, and not necessarily while driving but also while walking or cycling...

**Comment by sonora** [comment]
Agree. The only time a confirmation dialig should be **forced** upon you is when you are about to do something **destructive**, like maybe delete a track or similar, as this could be accidental.

Another example of a dangerous nag dialog while driving is: Be out driving with voice navigation on, now stop the navigation:
1. Tap the screen to get access to the navigation button
2. Tap the blue navigation button
3. Tap "Dismiss".
4. **Then you get another confirmation popup.**

(4) is totally obsolete. There seems low probability that you performed (1)-(3) in error. And there is nothing near the "Dismiss" button causing a mis-tap for (3). And even if you accidentally stopped your navigation, it is not 'destructive', since we have a 'last route' display now where you can restart it.

**Comment by pebogufi** [comment]
What is new since e few weeks: i can not edit the filename directly any more when stoping. It is no disaster but i am missing that. I use it on every stop/save.

**Comment by sonora** [comment]
Works for me, just tap into the file name text box of the dialog popping up after stopping the recording. What I find more annoying is that we save files even if they are empty.

**Comment by pebogufi** [comment]
I don't see a stop dialog with a file name. This the only stop dialog i see. After pressing "Beenden" there is nothing more.
Ap version 4.0.0#32139 2021-06-09

![StopDialog](https://user-images.githubusercontent.com/18628183/121800556-e9ed3480-cc32-11eb-8d03-bb0cc584d967.jpg)

**Comment by sonora** [comment]
Yes, I am talking about the dialog which pops up after tapping 'Beenden'.
![20210718_230802](https://user-images.githubusercontent.com/845735/126548463-58b425e7-79fa-49a0-b4de-420b43fdd64e.jpg)

**Comment by pebogufi** [comment]
Again: I don't get any dialog after "Beenden" on two different devices, both with app newer than 2021-06-10. 
Do you have any idea why this dialog with the filename is no more present ?
I am just traveling for several weeks, so i don't like to make a total UN- and reinstall. I need osmand, and i can accept smaler bugs or inconveniance rather than reinstall. 
Edit: is there any setting in json file of osf when exporting ?

**Reply by scaidermern** [reply]
Did you try a newer nightly version? Maybe there has been a bug which has been fixed meanwhile. I do get this dialog with the newest nightly.
